nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java

```diff
@@ -76,13 +77,28 @@
         return builder.explanation(reason).valid(reason == null).build();
     };
 
-    static final PropertyDescriptor QUERY = new PropertyDescriptor.Builder()
-            .name("Query")
-            .description("The selection criteria; must be a valid MongoDB Extended JSON format; if omitted the entire collection will be queried")
-            .required(false)
-            .expressionLanguageSupported(true)
-            .addValidator(DOCUMENT_VALIDATOR)
+    static final Relationship REL_SUCCESS = new Relationship.Builder().name("success").description("All files are routed to success").build();
+    static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("All input flowfiles that are part of a failed query execution go here.")
+            .build();
+
+    static final Relationship REL_ORIGINAL = new Relationship.Builder()
+            .name("original")
+            .description("All input flowfiles that are part of a successful query execution go here.")
             .build();
+
+    static final PropertyDescriptor QUERY = new PropertyDescriptor.Builder()
+        .name("Query")
+        .description("The selection criteria to do the lookup. If the field is left blank, it will look for input from" +
+                " an incoming connection from another processor to provide the query as a valid JSON document inside of " +
+                "the flowfile's body. If this field is left blank and a timer is enabled instead of an incoming connection, " +
```

Comment:
This needs to be reverted to the original description of behavior when blank, that the entire collection will be queried. If this is not the intent, and an error should be thrown, then you'll need to change the logic around line 264

Reply:
Done. Sorry about that.